ui/css/loading.scss

```diff
@@ -11,3 +13,37 @@
   align-self: center;
   margin-top: 1rem;
 }
+
+.loading-timeout-message {
+  align-self: center;
+  color: var(--color-text-alternative);
+  margin-top: 4rem;
+  max-width: design-system.$screen-sm-max;
+  text-align: center;
+
+  &::before {
+    direction: "__MSG_@@bidi_dir__";
+    content: "";
+    display: block;
+    white-space: pre-line;
+    animation: showTimeoutContent 20s step-end forwards;
+  }
+}
+
+@keyframes showTimeoutContent {
+  to {
+    // `\A` is a newline character.
+    content: "__MSG_CSS_loadingTakingTooLongMessageText__\A__MSG_CSS_loadingTakingTooLongActionText__";
```

Comment:
You can load messages from our `_locales/<lang>/message.json` files in CSS! Isn't that neat!

Note: this will use the user's global (OS/Browser) locale, not the MetaMask state's locale.